pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -267,10 +270,30 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		})
 	})
 	registerHandlers(c.pods.informer, c.queue, "Pods", c.pods.onEvent, nil)
-
+	if err := updateSPIFFERegistry(kubeClient, c.clusterID); err != nil {
+		log.Errorf("Failed to update the trust domain for cluster %v: %v", c.clusterID, err)
+	}
 	return c
 }
 
+func updateSPIFFERegistry(kubeClient kubelib.Client, clusterID string) error {
```

Comment:
Big problem here - it's a huge semantic change and security risk !!!!

We absolutely don't want to read mesh config from each and every registry ! Please move this code ( which is useful ) to the config controller, so only the 'main' cluster we read configs from handles MeshConfig.

Reply:
done.